tensorflow/core/kernels/crop_and_resize_op.cc

```diff
@@ -216,7 +217,8 @@ struct CropAndResize<CPUDevice, T> {
     const int depth = crops.dimension(3);
 
     // Sharding across boxes.
-    auto CropAndResizePerBox = [&](int start_box, int limit_box) {
```

Comment:
I meant making 2 lambda functions to be called inside `CropAndResizePerBox`, like this. Sorry for not being clear earlier!
```c++
    // Bilinear interpolation.
    auto interpolation_fn = [&](const float in_y) {
      const int top_y_index = floorf(in_y);
      const int bottom_y_index = ceilf(in_y);
      const float y_lerp = in_y - top_y_index;

      for (int x = 0; x < crop_width; ++x) {
        const float in_x = (crop_width > 1)
                               ? x1 * (image_width - 1) + x * width_scale
                               : 0.5 * (x1 + x2) * (image_width - 1);
        if (in_x < 0 || in_x > image_width - 1) {
          for (int d = 0; d < depth; ++d) {
            crops(b, y, x, d) = extrapolation_value;
          }
          continue;
        }
        const int left_x_index = floorf(in_x);
        const int right_x_index = ceilf(in_x);
        const float x_lerp = in_x - left_x_index;

        for (int d = 0; d < depth; ++d) {
          const float top_left(static_cast<float>(
              image(b_in, top_y_index, left_x_index, d)));
          const float top_right(static_cast<float>(
              image(b_in, top_y_index, right_x_index, d)));
          const float bottom_left(static_cast<float>(
              image(b_in, bottom_y_index, left_x_index, d)));
          const float bottom_right(static_cast<float>(
              image(b_in, bottom_y_index, right_x_index, d)));
          const float top = top_left + (top_right - top_left) * x_lerp;
          const float bottom =
              bottom_left + (bottom_right - bottom_left) * x_lerp;
          crops(b, y, x, d) = top + (bottom - top) * y_lerp;
        }
      }
    };
    if (method == "nearest") {  // Nearest-neighbor interpolation.
      interpolation_fn = [&](const float in_y) {
        for (int x = 0; x < crop_width; ++x) {
          const float in_x = (crop_width > 1)
                                 ? x1 * (image_width - 1) + x * width_scale
                                 : 0.5 * (x1 + x2) * (image_width - 1);
          if (in_x < 0 || in_x > image_width - 1) {
            for (int d = 0; d < depth; ++d) {
              crops(b, y, x, d) = extrapolation_value;
            }
            continue;
          }
          const int closest_x_index = roundf(in_x);
          const int closest_y_index = roundf(in_y);
          for (int d = 0; d < depth; ++d) {
            crops(b, y, x, d) = static_cast<float>(
                image(b_in, closest_y_index, closest_x_index, d));
          }
        }
      };
    }
    auto CropAndResizePerBox = [&](int start_box, int limit_box) {
      for (int b = start_box; b < limit_box; ++b) {
        const float y1 = boxes(b, 0);
        const float x1 = boxes(b, 1);
        const float y2 = boxes(b, 2);
        const float x2 = boxes(b, 3);

        const int32 b_in = box_index(b);
        if (!FastBoundsCheck(b_in, batch_size)) {
          continue;
        }

        const float height_scale =
            (crop_height > 1)
                ? (y2 - y1) * (image_height - 1) / (crop_height - 1)
                : 0;
        const float width_scale =
            (crop_width > 1) ? (x2 - x1) * (image_width - 1) / (crop_width - 1)
                             : 0;

        for (int y = 0; y < crop_height; ++y) {
          const float in_y = (crop_height > 1)
                                 ? y1 * (image_height - 1) + y * height_scale
                                 : 0.5 * (y1 + y2) * (image_height - 1);
          if (in_y < 0 || in_y > image_height - 1) {
            for (int x = 0; x < crop_width; ++x) {
              for (int d = 0; d < depth; ++d) {
                crops(b, y, x, d) = extrapolation_value;
              }
            }
            continue;
          }
          interpolation_fn(in_y);
        }
      }
    };
```